WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt

```diff
@@ -3,26 +3,70 @@ package org.wordpress.android.ui.jetpack.scan
 import android.os.Bundle
 import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
+import kotlinx.android.synthetic.main.scan_fragment.recycler_view
 import org.wordpress.android.R
 import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
+import org.wordpress.android.ui.jetpack.scan.adapters.ScanAdapter
+import org.wordpress.android.ui.utils.UiHelpers
 import javax.inject.Inject
 
 class ScanFragment : Fragment(R.layout.scan_fragment) {
+    @Inject lateinit var uiHelpers: UiHelpers
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     private lateinit var viewModel: ScanViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        (requireActivity().application as WordPress).component().inject(this)
-    }
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        initViewModel()
+        initDagger()
+        initAdapter()
+        initViewModel(getSite(savedInstanceState))
+    }
+
+    private fun initDagger() {
+        (requireActivity().application as WordPress).component()?.inject(this)
     }
 
-    private fun initViewModel() {
+    private fun initAdapter() {
+        val scanAdapter = ScanAdapter(uiHelpers)
+        scanAdapter.setHasStableIds(true)
+        recycler_view.adapter = scanAdapter
+    }
+
+    private fun initViewModel(site: SiteModel) {
         viewModel = ViewModelProvider(this, viewModelFactory).get(ScanViewModel::class.java)
+        setupObservers()
+        viewModel.start(site)
+    }
+
+    private fun setupObservers() {
+        viewModel.uiState.observe(
+            viewLifecycleOwner,
+            Observer { uiState ->
+                if (uiState is Content) {
+                    refreshContentScreen(uiState)
+                }
+            }
+        )
+    }
+
+    private fun refreshContentScreen(content: Content) {
+        ((recycler_view.adapter) as ScanAdapter).update(content.items)
+    }
+
+    private fun getSite(savedInstanceState: Bundle?): SiteModel {
```

Comment:
I haven't quite understood this comment. Let's discuss it tomm. when you're online.

Reply:
This was not an important comment so I'll just close this convo to unblock this PR.

The first part `Have you considered using a field instead of this method? It feels that the method isn't re-usable. Since if I want to invoke it eg. from "observe(..)" I don't have access to the "savedInstanceState" anymore. Wdyt?` The current usage of the method is completely fine and I don't see any issues with it. However, if you imagine that we'd like to invoke `getSite(..)` from a different place than `onViewCreated()` we won't have access to `savedInstanceState`.  So we might want to consider introducing "site" field rather than using this method.